### ISSUE TITLE
Run in terminal command edits the current file instead of running in terminal

### ISSUE THREAD

**Issue by fbricon** [continuedev/continue#3045]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: macOS Sequoia 15.1 (24B83)
- Continue version: built from source (aa59140a82)
- IDE version: VS Code 1.95.3
- Model: granite3-dense:8b
- config.json:
  
{
    "models": [
        {
            "model": "granite3-dense:8b",
            "provider": "ollama",
            "contextLength": 4096,
            "completionOptions": {
                "maxTokens": 2048,
                "temperature": 0,
                "topP": 0.9,
                "topK": 40,
                "presencePenalty": 0,
                "frequencyPenalty": 0.1
            },
            "systemMessage": "You are Granite, an AI language model developed by IBM. You are a cautious assistant. You carefully follow instructions. You are helpful and harmless and you follow ethical guidelines and promote positive behavior.",
            "title": "granite3-dense:8b"
        }
    ],
    "tabAutocompleteModel": {
        "model": "granite3-dense:2b",
        "provider": "ollama",
        "contextLength": 4096,
        "completionOptions": {
            "maxTokens": 2048,
            "temperature": 0,
            "topP": 0.9,
            "topK": 40,
            "presencePenalty": 0,
            "frequencyPenalty": 0.1
        },
        "title": "granite3-dense:2b"
    },
    "tabAutocompleteOptions": {
        "useCache": false
    },
    "customCommands": [
        {
            "name": "test",
            "prompt": "{{{ input }}}\n\nWrite a comprehensive set of unit tests for the selected code. It should setup, run tests that check for correctness including important edge cases, and teardown. Ensure that the tests are complete and sophisticated. Give the tests just as chat output, don't edit any file.",
            "description": "Write unit tests for highlighted code"
        }
    ],
    "contextProviders": [
        {
            "name": "code",
            "params": {}
        },
        {
            "name": "docs",
            "params": {}
        },
        {
            "name": "diff",
            "params": {}
        },
        {
            "name": "terminal",
            "params": {}
        },
        {
            "name": "problems",
            "params": {}
        },
        {
            "name": "folder",
            "params": {}
        },
        {
            "name": "codebase",
            "params": {}
        }
    ],
    "slashCommands": [
        {
            "name": "edit",
            "description": "Edit selected code"
        },
        {
            "name": "comment",
            "description": "Write comments for the selected code"
        },
        {
            "name": "share",
            "description": "Export the current chat session to markdown"
        },
        {
            "name": "cmd",
            "description": "Generate a shell command"
        },
        {
            "name": "commit",
            "description": "Generate a git commit message"
        }
    ],
    "embeddingsProvider": {
        "provider": "ollama",
        "model": "nomic-embed-text:latest",
        "title": "nomic-embed-text:latest"
    }
}
```

### Description

The "Run in Terminal" command from the chat window, when clicked, sends an edit request to the LLM and then modifies the currently opened file, instead of running the command in a terminal

https://github.com/user-attachments/assets/5ae54ab9-97e6-45ff-ad41-8cbde8814225

### To reproduce

1. open a new folder
2. create a main.py file
3. open the model chat view
4. select granite3-dense:8b
5. Execute the following prompt : "Create a Python REST API that talks to LLM"
6. Apply the proposed code
7. on the `pip install flask request`, click "Run in terminal"

The main.py code is being deleted, 

### Log output

```Shell
the Continue LLM Prompt output view shows:

> ##### Prompt #####
> 
> from flask import Flask, request, jsonify
> import requests
> app = Flask(__name__)
> @app.route('/generate_text', methods=['POST'])
> def generate_text():
>     data = request.get_json()
>     prompt = data['prompt']
>     # Replace 'https://api.langchain.com/generate' with the actual LLM API endpoint
>     response = requests.post('https://api.langchain.com/generate', json={'prompt': prompt})
>     generated_text = response.json()['generated_text']
>     return jsonify({'generated_text': generated_text})
> if __name__ == '__main__':
>     app.run(debug=True)
> 
> Please rewrite the above file to address the following request:
> The following code was suggested as an edit:
> ```
> pip install flask requests
> ```
> Please apply it to the previous code.
```

**Comment by fbricon** [comment]
It looks like the Run in Terminal command triggered an apply edit for the next code block (`python app.py`, not recognized as a command itself)

**Comment by fbricon** [comment]
Well, even if the last command the LLM returns is 

> ```bash
> pip install flask requests
> ```

it still tries to edit the file instead of executing it in a terminal

**Comment by fbricon** [comment]
I tracked down the regression to PR #2861. The code to run in a terminal : 
https://github.com/continuedev/continue/pull/2861/files#diff-37c238ae74065c74516ebd4fcd159f7af0e30cc4605c80ef9366b1a4dc29e2e5L176-L179

was lost in the refactoring to:

https://github.com/continuedev/continue/pull/2861/files#diff-6e0c37673c3f439ee9b319f6e3688406b19ade2a2e1fce9272df6646b1b8c664R16-R42

cc @Patrick-Erichsen

**Comment by fbricon** [comment]
See proposed fix: #3050
